src/execution.rs

```diff
@@ -263,6 +270,29 @@ impl ExecutionError {
     }
 }
 
+/// Trait to signal that the execution is cancelled
+pub trait CancellationFlag {
+    fn check(&self, at: &'static str) -> Result<(), CancellationError>;
+}
+
+pub struct CancellationFlags;
+impl CancellationFlags {
+    pub fn none() -> impl CancellationFlag {
+        struct NoCancellation;
+        impl CancellationFlag for NoCancellation {
+            fn check(&self, _at: &'static str) -> Result<(), CancellationError> {
+                Ok(())
+            }
+        }
+        NoCancellation
+    }
+}
```

Comment:
You could expose `NoCancellation` directly instead of having the factory function.  That would change callers from `&CancellationFlags::none()` to `&NoCancellation`.  I have a slight aesthetic preference for exposing the type directly, unless there's something that I'm missing.

Reply:
In an earlier iteration `NoCancellation` had parameters and this was a neat way to hide all that from the user. I would have preferably written `none()` as an associated function of `CancellationFlag` but that is not technically possible.

I don't think there is much of a difference between the two options now, so I'm fine either way.